wpilibj/src/main/java/edu/wpi/first/wpilibj/RobotBase.java

```diff
@@ -285,6 +274,33 @@ public static boolean getBooleanProperty(String name, boolean defaultValue) {
         DriverStation.reportError("Unexpected return from startCompetition() method.", false);
       }
     }
+  }
+
+  /**
+   * Starting point for the applications.
+   */
+  public static <T extends RobotBase> void startRobot(Supplier<T> robotSupplier) {
+    if (!HAL.initialize(500, 0)) {
+      throw new IllegalStateException("Failed to initialize. Terminating");
+    }
+
+    // Call a CameraServer JNI function to force OpenCV native library loading
+    // Needed because all the OpenCV JNI functions don't have built in loading
+    CameraServerJNI.enumerateSinks();
+
+    HAL.report(tResourceType.kResourceType_Language, tInstances.kLanguage_Java);
+
+    if (HAL.hasMain()) {
+      Thread thread = new Thread(() -> {
+        runRobot(robotSupplier);
+        HAL.exitMain();
+      }, "robot main");
+      thread.setDaemon(true);
+      thread.start();
```

Comment:
This is just going to exit instantly if there is a HAL main? Does there need to be a wait after starting the thread?

Reply:
Whoops!  Fixed